projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -6677,14 +6677,16 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
      */
     protected autogenerateColumns() {
         const data = this.gridAPI.get_data();
+        const factory = this.resolver.resolveComponentFactory(IgxColumnComponent);
         const fields = this.generateDataFields(data);
         const columns = [];
 
         fields.forEach((field) => {
-            const newColumn = new IgxColumnComponent(this, this.cdr, this.platform);
-            newColumn.field = field;
-            newColumn.dataType = this.resolveDataTypes(data[0][field]);
-            columns.push(newColumn);
+            const ref = factory.create(this.viewRef.injector);
```

Comment:
You can still remove the deprecated factory instantiation:
```typescript
const ref = this.viewRef.createComponent(IgxColumnComponent)
```

Reply:
@rkaraivanov I'll work on this, because removing the deprecated factory is a bit larger in terms of a change, because it's used in other places as well.

Reply:
@rkaraivanov I decided to remove it completely in a separate PR.